indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/TasksAndWorkersFilteredByIp.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord.autoscaling;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import io.druid.indexing.common.task.Task;
+import io.druid.indexing.overlord.ImmutableWorkerInfo;
+import io.druid.indexing.overlord.TaskRunner;
+import io.druid.indexing.overlord.TaskRunnerWorkItem;
+import io.druid.indexing.overlord.TasksAndWorkers;
+import io.druid.indexing.overlord.WorkerTaskRunner;
+import io.druid.indexing.overlord.config.WorkerTaskRunnerConfig;
+import io.druid.indexing.worker.Worker;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+public class TasksAndWorkersFilteredByIp implements TasksAndWorkers
+{
+  private final WorkerTaskRunner delegate;
+  private final String ipPrefix;
+  private final Predicate<Task> taskPredicate;
+
+  public TasksAndWorkersFilteredByIp(WorkerTaskRunner delegate, String ipPrefix, Predicate<Task> taskPredicate)
+  {
+    this.delegate = delegate;
+    this.ipPrefix = ipPrefix;
+    this.taskPredicate = taskPredicate;
+  }
+
+  private Collection<Worker> filter(Collection<Worker> workers)
+  {
+    Collection<Worker> filtered = new ArrayList<>(workers.size());
```

Comment:
Couldn't be done without anonymous lambda, that is much more cumbersome before Java 8 than the present code